Dynatrace/dynatrace-operator
# Fix data-ingests's `endpoint.properties` secret handling

# Description
1. Secret `dynatrace-data-ingest-endpoint` is now always generated nevertheless which ingestions are enabled.
If no ingestion is enabled, secret's data contains an empty string, as expected.

2. Fixed bug causing permanent reconciliation which was caused by comparing `nil` and an empty map, assuming it's equal.

# How can this be tested?
Check if `dynatrace-data-ingest-endpoint` secret in `deafult` namespace contains expected content.

Dynakubes to be used:
1. default one, with no annotation
2. with AG `statsd-ingest` capability enabled

Maniulate dynakubes with:
1. `k -ndynatrace annotate dynakubes.dynatrace.com dk --overwrite feature.dynatrace.com/disable-metadata-enrichment=false` - no change should be observed
2. `k -ndynatrace annotate dynakubes.dynatrace.com dk --overwrite feature.dynatrace.com/disable-metadata-enrichment=true` - metrics ingest URL and token should be removed from the secret data

## Checklist
